library/core/src/array/drain.rs

```diff
@@ -39,6 +40,8 @@ impl<T> Drop for Drain<'_, T> {
 }
 
 impl<T> Iterator for Drain<'_, T> {
+    #[cfg_attr(not(bootstrap), allow(unnameable_types))]
+    //~^ reachable at visibility `pub(crate)`, but can only be named at visibility `pub(array)`
     type Item = T;
```

Comment:
It looks like we should stop reporting this lint for associated types.
All the cases reported in this PR are false positives / not useful, and I don't see when it would be useful to report this for associated types in general.